eyelid-match-ops/src/primitives/hamming.rs

```diff
@@ -0,0 +1,125 @@
+//! Implementation of the simple encoding
+
+use crate::primitives::poly::Poly;
+use ark_ff::Zero;
+use std::ops::{AddAssign, Sub};
+
+use rand::rngs::ThreadRng;
+
+use super::yashe::Yashe;
+use super::yashe::{Ciphertext, Message, PrivateKey, PublicKey, YasheConf};
+
+/// Contains the message to be encoded such that
+/// the Hamming distance can be computed later.
+pub struct SimpleHammingEncoding<C: YasheConf>
+where
+    C::Coeff: From<u128> + From<u64> + From<i64>,
+{
+    /// The message to be encoded
+    m: Message<C>,
+    /// The reverse of the message to be encoded
+    m_rev: Message<C>,
+}
+
+/// SimpleHammingEncodingCiphertext is a struct that holds two ciphertexts, c and c_rev,
+/// which are the encodings of the message m and m_rev, respectively. The encoding is
+/// done by reversing the message and encoding it as a regular Yashe Ciphertext.
+pub struct SimpleHammingEncodingCiphertext<C: YasheConf>
+where
+    C::Coeff: From<u128> + From<u64> + From<i64>,
+{
+    /// The ciphertext of the message m
+    c: Ciphertext<C>,
+    /// The ciphertext of the message m_rev
+    c_rev: Ciphertext<C>,
+}
+
+impl<C: YasheConf> SimpleHammingEncoding<C>
+where
+    C::Coeff: From<u128> + From<u64> + From<i64>,
+{
+    /// Creates a new `SimpleHammingEncoding` with the given message `m` and size `size`.
+    pub fn new(m: Message<C>, size: usize) -> Self {
```

Comment:
In this file, I think `size` is always `<C as PolyConf>::MAX_POLY_DEGREE`?

Reply:
Not necessarily. The user could be interested in any `size` that is smaller than `MAX_POLY_DEGREE`, so they could embed their vector in some polynomial configuration, maybe wasting some coefficients that are not used. I tested with 1000 to emphasize that this can be chosen by the user. This is important because the encoding depends on this value of `size`. When we revert the coefficient, we do it inside the sub-vector that has `size` elements, which is different from reverting all the coefficients of the polynomial.

Reply:
```suggestion
    ///
    /// `size` can be any value less than MAX_POLY_DEGREE. This allows vectors to be embedded
    /// in any polynomial configuration, possibly wasting some coefficients that are not used.
    /// The encoding depends on this value of size. When the coefficient is reverted, this happens
    /// inside the sub-vector that has `size` elements, which is different from reverting all the
    /// coefficients of the polynomial.
    pub fn new(m: Message<C>, size: usize) -> Self {
```